server/src/main/java/org/apache/cassandra/sidecar/utils/SidecarClientProvider.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.cassandra.sidecar.client.SimpleSidecarInstancesProvider;
+import org.apache.cassandra.sidecar.client.VertxHttpClient;
+import org.apache.cassandra.sidecar.client.retry.ExponentialBackoffRetryPolicy;
+import org.apache.cassandra.sidecar.client.retry.RetryPolicy;
+import org.apache.cassandra.sidecar.common.client.SidecarInstance;
+import org.apache.cassandra.sidecar.common.client.SidecarInstanceImpl;
+import org.apache.cassandra.sidecar.common.server.utils.SidecarVersionProvider;
+import org.apache.cassandra.sidecar.config.SidecarClientConfiguration;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+
+/**
+ * Provider class for retrieving the singleton {@link SidecarClient} instance
+ */
+@Singleton
+public class SidecarClientProvider implements Provider<SidecarClient>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(SidecarClientProvider.class);
+    private final Vertx vertx;
+    private final SidecarClientConfiguration clientConfig;
+    private final SidecarVersionProvider sidecarVersionProvider;
+    private final SidecarClient client;
+
+    @Inject
+    public SidecarClientProvider(Vertx vertx,
+                                 SidecarConfiguration sidecarConfiguration,
+                                 SidecarVersionProvider sidecarVersionProvider)
+    {
+        this.vertx = vertx;
+        this.clientConfig = sidecarConfiguration.sidecarClientConfiguration();
+        this.sidecarVersionProvider = sidecarVersionProvider;
+        this.client = initializeSidecarClient();
+    }
+
+    @Override
+    public SidecarClient get()
+    {
+        return client;
+    }
+
+    public void close(Promise<Void> completion)
+    {
+        LOGGER.info("Closing Sidecar Client...");
+        try
+        {
+            client.close();
+            Thread.sleep(100);
+            completion.complete();
+        }
+        catch (Throwable throwable)
+        {
+            completion.fail(throwable);
+        }
+    }
+
+    private SidecarClient initializeSidecarClient()
+    {
+        WebClientOptions webClientOptions = webClientOptions();
+        HttpClient httpClient = vertx.createHttpClient(webClientOptions);
+        WebClient webClient = WebClient.wrap(httpClient, webClientOptions);
+
+        HttpClientConfig httpClientConfig = new HttpClientConfig.Builder<>()
+                                            .ssl(webClientOptions().isSsl())
+                                            .timeoutMillis(clientConfig.requestTimeout().toMillis())
+                                            .idleTimeoutMillis(clientConfig.requestIdleTimeout().toIntMillis())
+                                            .userAgent("cassandra-sidecar/" + sidecarVersionProvider.sidecarVersion())
+                                            .build();
+
+        VertxHttpClient vertxHttpClient = new VertxHttpClient(vertx, webClient, httpClientConfig);
```

Comment:
If we add missing settings in `HttpClientConfig` we dont have to create `WebClient` separately, we can directly call `VertxHttpClient vertxHttpClient = new VertxHttpClient(vertx, httpClientConfig)`